packages/axios-extension/src/abap/lrep-service.ts

```diff
@@ -0,0 +1,163 @@
+import type { Service } from '../base/service-provider';
+import type { AxiosResponse } from 'axios';
+import { Axios } from 'axios';
+import { LogLevel } from '@sap-ux/logger';
+import type { Logger } from '@sap-ux/logger';
+import { readFileSync } from 'fs';
+
+/**
+ * Type representing a namespace. It is either a string or an object containing an id (variant id) and a reference (base application id).
+ */
+export type Namespace =
+    | string
+    | {
+          /**
+           * variant id
+           */
+          id: string;
+
+          /**
+           * base application id
+           */
+          reference: string;
+      };
+
+/**
+ * Required configuration to deploy an adaptation project.
+ */
+export interface AdaptationConfig {
+    /**
+     * Namespace either as string or object
+     */
+    namespace: Namespace;
+
+    /**
+     * Optional ABAP package name
+     */
+    package?: string;
+
+    /**
+     * Optional transport request
+     */
+    transport?: string;
+}
+
+/**
+ * Structure of the result message.
+ */
+export interface Message {
+    severity: 'Success' | 'Warning' | 'Error';
+    text: string;
+    details?: string[];
+    id: string;
+    variables?: string[];
+}
+
+/**
+ * Technically supported layers, however, in practice only `CUSTOMER_BASE` is used
+ */
+type Layer = 'VENDOR' | 'CUSTOMER_BASE';
+
+/**
+ * Returns the namespace as string.
+ *
+ * @param namespace either as string or as object
+ * @returns serialized namespace
+ */
+function getNamespaceAsString(namespace: Namespace): string {
+    return namespace['id'] ? `apps/${namespace['reference']}/appVariants/${namespace['id']}/` : (namespace as string);
+}
+
+/**
+ * A class respresenting the design time adaptation service allowing to deploy adaptation projects to an ABAP system.
+ */
+export abstract class LayeredRepositoryService extends Axios implements Service {
+    public static readonly PATH = '/sap/bc/lrep';
+
+    public log: Logger;
+
+    /**
+     * Check whether a variant with the given namespace already exists.
+     *
+     * @param namespace either as string or as object
+     * @returns the Axios response object for futher processing
+     */
+    public async check(namespace: Namespace): Promise<AxiosResponse> {
```

Comment:
Does this method need to be public? If so, does it make sense to wrap the raw axios response?

Reply:
Does it need to be, no, but the guiding principle for the axios-extension is to offer convenience but still keep it flexible, therefore, I would like to keep it public and also return the axios response.

The naming I will change.

Reply:
Fixed with https://github.com/SAP/open-ux-tools/pull/469/commits/632872d952a710b29d2a0114a841ddd0d88a4f58